conda_build/metadata.py

```diff
@@ -88,6 +88,26 @@ def ns_cfg(config):
 # - (?(2).*)$ means "allow trailing characters iff group 2 (#.*) was found."
 sel_pat = re.compile(r'(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2).*)$')
 
+# this function extracts the variable name from a NameError exception, it has the form of:
+# "NameError: name 'var' is not defined", where var is the variable that is not defined. This gets returned
+def parseNameNotFound(error):
+    m = re.search('\'(.+?)\'', str(error))
+    if len(m.groups()) == 1:
+        return m.group(1)
+    else:
+        return ""
+
+# We evaluate the selector and return True (keep this line) or False (drop this line)
+# If we encounter a NameError (unknown variable in selector), then we replace it by False and re-run the evaluation
+def eval_selector(selector_string, namespace):
+    try:
+        # TODO: is there a way to do this without eval?  Eval allows arbitrary
+        #    code execution.
+        return eval(selector_string, namespace, {})
+    except NameError as e:
+        missing_var = parseNameNotFound(e)
+        next_string = selector_string.replace(missing_var, "False")
+        return eval_selector(next_string, namespace)
```

Comment:
Please show a warning log message at least, so misspellings are not just always silent failures.